### PR TITLE
fix: overlapping content across catalogs constantly causing recreates

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
[related ticket](https://2u-internal.atlassian.net/browse/ENT-5849)

TLDR- we noticed that certain pieces of content were constantly being "created" (or attempted to be) for customers through the integrated channels. The common factor between all these customers and all this content was that the customers had multiple catalogs that shared the content.

**The problem:** 
The integrated channels (roughly) work as such-
- we iterate over each customer catalog
- we collect all transmission audits under each catalogs, one by one
   - specifically we filter the audit table by customer + catalog uuid
- we take the collected audits and send them to the ent catalog service and ask what work needs to be done
- we take the work that needs to be done and send it to the customer
- after successfully sending a piece of content to the customer, we save the record as a transmission audit
   - IF THERE EXISTS A RECORD OF THAT PIECE OF CONTENT FOR THE CUSTOMER UNDER A DIFFERENT CATALOG, we edit the record instead of creating a new one, swapping the catalog UUID to the new catalog. 

with this logic, on the second transmit content metadata run (and subsequently all future runs), the content will be missing under the list of sent content of one of the overlapping catalogs when we go to fetch all previous transmission audits. Thus when we ask the ent catalog service what works needs to be done, it will report that the content needs to be created. This is what creates the endless stream of "creating content X" for customers.

**The fix**
Unfortunately when collecting all audits, we can't look up content across catalogs. Due to how the catalog diff endpoint functions, we have to report content/ask for work to do catalog by catalog. Therefore the fix is to move down the line- when we get the list of work from the catalog diff endpoint, we have to check any requested creates to see if we have a record of that to-be-created content already existing on the customer's instance.   If it does already exist then we simply remove it from the list of things to do.

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: integrated_channels/integrated_channel/exporters/content_metadata.py [hunk 1]
```diff
@@ -169,12 +176,32 @@ def _get_catalog_diff(self, enterprise_catalog, content_keys, force_retrieve_all
...
+        ContentMetadataItemTransmission = apps.get_model(
+            'integrated_channel',
+            'ContentMetadataItemTransmission'
+        )
+        unique_new_items_to_create = []
+
+        # We need to remove any potential create transmissions if the content already exists on the customer's instance
+        # under a different catalog
+        for index, item in enumerate(items_to_create):
+            item_created_in_other_catalog = ContentMetadataItemTransmission.objects.filter(
```
Comment (mattdrayer) [inline integrated_channels/integrated_channel/exporters/content_metadata.py:189]: This seems like a lot of database I/O work -- does it make more sense to simply pull the full set of audit record content_ids for the customer UUID, that are not "deleted" and then reconcile with that list?

Comment (alex-sheehan-edx) [reply integrated_channels/integrated_channel/exporters/content_metadata.py:189]: good point!

**Review by mattdrayer — APPROVED** [review]
LGTM!  We've discussed offline about ways to improve on the `for` loop implementation, which could be handled as a follow-on enhancement.  However, given that this will typically execute as an asynchronous task on a pool worker it's less of a concern.

**Review by bcitro — APPROVED** [review]
🚢

**Review by johnnagro — APPROVED** [review]
⛵
